### DISCUSSION TITLE
LazyTerm to open lf and use it as a file explorer

### DISCUSSION THREAD

**Discussion by msMatix** [LazyVim/LazyVim#1944] [Q&A]
Hello everyone :),

I am currently trying to use lf (see: https://github.com/gokcehan/lf) similar to Lazygit and have made the following keybinding for it:

```
vim.keymap.set("n", "<leader>gl", function()
  Util.terminal({ "lf", "-selection-path", "/dev/stdout" }, { cwd = Util.root(), esc_esc = false, ctrl_hjkl = false })
end, { desc = "LF (root dir)" })
```

Now I want to use lf to select the file I want to open. Similar to the standard file explorer in LazyVim. I do the whole thing with the cmd: `lf -selection-path /dev/stdout `

The result gives me the selection I would like to call when closing lf. With Toggleterm there is the functionality `on_exit`. Is this also available for the LazyTerm? Unfortunately, I haven't found anything about it.

Thank you very much!

Matix

**Comment by dpetka2001** [comment]
Not quite a direct answer to your question, but have you looked at [lf.vim](https://github.com/ptzz/lf.vim)? Alternatively, to emulate the Toggleterm `on_exit` function, maybe you could create an autocommand on event `TermClose` to do what you want?

**Comment by msMatix** [comment]
yes I did. But I thought it might be nice to keep the amount of plugins low. Ah the autocommand is a nice I idea. I will try this. 

Thanks for the tip.
